Selenium.Axe/Selenium.Axe/HtmlReport.cs

```diff
@@ -0,0 +1,249 @@
+using System.Web;
+
+namespace Selenium.Axe
+{
+    public static class HtmlReport
+    {
+        public static void CreateReport(this IWebDriver webDriver, string destination)
+        {
+            AxeBuilder axeBuilder = new AxeBuilder(webDriver);
+            webDriver.CreateReport(axeBuilder.Analyze(), destination);
+        }
+
+        public static void CreateReport(this IWebDriver webDriver, IWebElement context, string destination)
+        {
+            AxeBuilder axeBuilder = new AxeBuilder(webDriver);
+            context.CreateReport(axeBuilder.Analyze(context), destination);
+        }
+        public static void CreateReport(this ISearchContext context, AxeResult results, string destination)
+        {
+             HashSet<string> selectors = new HashSet<string>();
+            int violationCount = GetCount(results.Violations, ref selectors);
+            int incompleteCount = GetCount(results.Incomplete, ref selectors);
+            int passCount = GetCount(results.Passes, ref selectors);
+            int inapplicableCount = GetCount(results.Inapplicable, ref selectors);
+
+            var doc = new HtmlDocument();
+
+            var node = HtmlNode.CreateNode("<html lang=\"en\"><head><meta charset=\"utf-8\"><title>Accessiblity Check</title><style></style></head><body></body></html>");
+            doc.DocumentNode.AppendChild(node);
+
+            HtmlCommentNode hcn = doc.CreateComment("<!DOCTYPE html>\r\n");
+            HtmlNode htmlNode = doc.DocumentNode.SelectSingleNode("/html");
+            doc.DocumentNode.InsertBefore(hcn, htmlNode);
+
+            StringBuilder content = new StringBuilder();
+            content.AppendLine(@".fullImage{");
+            content.AppendLine($"content: url('{GetDataImageString(context)};border: 1px solid black;margin-left:1em;");
+            content.AppendLine(@"}
+.fullImage:hover {transform:scale(2.75);transform-origin: top left;}
+p {}
+.wrap .wrapTwo .wrapThree{overflow-wrap: anywhere;margin: 2px;}
+.wrapOne {margin-left:1em;}
+.wrapTwo {margin-left:2em;}
+.wrapThree {margin-left:3em;}
+.emOne {margin-left:1em;}
+.emTwo {margin-left:2em;}
+.emThree {margin-left:3em;}
+.majorSection{border: 1px solid black;}
+.findings{border-top:1px solid black;}
+.htmlTable{border-color:lightgray;border-style:solid;width:100%;display:table;}");
+
+            HtmlNode body = doc.DocumentNode.SelectSingleNode("//body");
+            doc.DocumentNode.SelectSingleNode("//style").InnerHtml = content.ToString();
+
+            var element = doc.CreateElement("h1");
+            element.InnerHtml = "Accessiblity Check";
+            body.AppendChild(element);
+
+            element = doc.CreateElement("h3");
+            element.InnerHtml = "Context:";
+            body.AppendChild(element);
+
+            content = new StringBuilder();
+            content.AppendLine($"Url: {results.Url}<br>");
+            content.AppendLine($"Orientation: {results.TestEnvironment.OrientationType}<br>");
```

Comment:
I like including the test environment context in the report; I'd recommend also including window size (we regularly see accessibility issues that only repro at certain breakpoints)

Reply:
Good call, adding